### ISSUE TITLE
Errors when trying to use locally built libwebrtc.a files from libwebrtc-bin

### ISSUE THREAD

**Issue by feniljain** [crow-misia/libmediasoup-android#6]
Hey @crow-misia,

### AIM:

I want to build `libmediasoup-android` with the latest version of `libmediasoupclient` and latest commit on 4606(m94) branch of `libwebrtc` i.e. commits `8b36a91520a0f6ea3ed506814410176a9fc71d62` and `6584528aeb0f0e2ab4d14114aefeee7e5997ade9` respectively.

### TRY:

To achieve this, I tried to locally build binaries of libwebrtc and use them in current project. I was trying to build against the latest version of 4606 branch(m94). I took the built binaries and dropped them into `core/deps/webrtc/` and built `libmediasoup-android`. Though it failed with errors mentioning:

[logs-65845.txt](https://github.com/crow-misia/libmediasoup-android/files/8095482/logs-65845.txt)

Then I decided to once experiment with build in releases page(the one I used was `94.4606.4.1`), which worked perfectly(with latest mediasoup version too). Then I built that same commit(`81f4af57823c28c903b61b273e6fc2b87787b11b`) locally and tried use it. Which to my suprise gave these errors:

[logs-81f4a57.txt](https://github.com/crow-misia/libmediasoup-android/files/8095520/logs-81f4a57.txt)

The steps I am following to build it locally are:

- `make -C build android`
- Then take the tar and uncompress it in `core/deps/webrtc/`

Is there something I am missing while building it locally?

### SOME OBSERVATIONS:

From `logs-81f4a57.txt`, I noticed a line mentioning:

```
error: no template named 'is_same_v' in namespace 'std'; did you mean 'is_same'?
```

On further investigation upon this I realized this was cpp17 vs cpp14 issue, but in webrtc docs it's clearly mentioned that they use cpp 14 itself and even in the CMakeLists.txt of this project, min dependency of cpp14 is required, that confused me what could be causing this error.

Finally I decided to file an issue to get your thoughts on this, my end goal still remains the same. Any help would be appreciated!

P.S.: I was confused whether to file an issue on this repo or `libwebrtc-bin` though it also involves building `libmediasoup-android` I decided to put it here.

**Comment by crow-misia** [comment]
libwebrtc commit
81f4af57823c28c903b61b273e6fc2b87787b11b is m99.4844@{#1}, not m94.4606{#4}.

libwebrtc's commit for m94.4606{#4} is 6584528aeb0f0e2ab4d14114aefeee7e5997ade9.

Prebuild binary of 99.4844.1.0 in libwebrtc-bin repository is configured to use custom libcxx, so you cannot build it just by replacing a version.

**Comment by feniljain** [comment]
Hey,

Yup I was wondering the same why the commit was much later's, but in your releases page I saw release of m94 having a commit mentioning this commit from webrtc:

https://github.com/crow-misia/libwebrtc-bin/blob/9d7cfdd14b6915a6d0b5e0076ad1269802315c60/VERSION#L4

I presumed you made some changes from your side. Thanks for clarifying the misassumption.

With that out of the way, I did try to build `libmediasoup-android` with binaries built with this commit hash:
`6584528aeb0f0e2ab4d14114aefeee7e5997ade9`

But I quickly encountered an error whose log is given the main issue text with filename: `logs-65845.txt`.

Any ideas what could be going wrong in that?
